### PR TITLE
Update Room Documentation

### PR DESCRIPTION
**Pull request by mzxrules**
Sister pr to https://github.com/zeldaret/oot/pull/2015

The only difference is that I didn't attempt to name the last few variable since I'm less confident on those.

### CODE REVIEW

File: include/z64scene.h [hunk 1]
```diff
@@ -348,11 +347,10 @@ typedef struct {
...
     /* 0x14 */ Room prevRoom;
-    /* 0x28 */ void* roomMemPages[2]; // In a scene with transitions, roomMemory is split between two pages that toggle each transition. This is one continuous range, as the second page allocates from the end
-    /* 0x30 */ u8 activeMemPage; // 0 - First page in memory, 1 - Second page
-    /* 0x31 */ s8 status;
-    /* 0x32 */ UNK_TYPE1 pad32[0x2];
-    /* 0x34 */ void* activeRoomVram;
+    /* 0x28 */ void* bufPtrs[2]; // Start and end pointers for the room buffer. Can be split into two pages, where page 0 is allocated from the start pointer and page 1 is allocated from the end pointer.
+    /* 0x30 */ u8 activeBufPage; // 0 - First page in memory, 1 - Second page
+    /* 0x31 */ s8 status; // 0 - Free for new room request, 1 - DmaRequest for a new room is in progress
+    /* 0x34 */ void* roomRequestAddr; // ram pointer to where the requested room segment will be stored
```
Comment (AngheloAlf) [inline include/z64scene.h:353]: "ram pointer" is redundant, isn't it?
```suggestion
    /* 0x34 */ void* roomRequestAddr; // Pointer to where the requested room segment will be stored
```

**Review by AngheloAlf — COMMENTED** [review]
one nit

File: include/z64scene.h [hunk 2]
```diff
@@ -362,6 +360,11 @@ typedef struct RoomContext {
     /* 0x7A */ UNK_TYPE2 unk7A[3];
 } RoomContext; // size = 0x80
 
+typedef struct RoomList {
+    /* 0x00 */ u8 count;
+    /* 0x04 */ RomFile* romFiles; // Array of rom addresses for each room in a scene
+} RoomList;
```
Comment (hensldm) [inline include/z64scene.h:366]:
```suggestion
typedef struct RoomList {
    /* 0x0 */ u8 count;
    /* 0x4 */ RomFile* romFiles; // Array of rom addresses for each room in a scene
} RoomList; // size = 0x8
```
remove trailing 0's please, as well as add a size comment.

**Review by hensldm — COMMENTED** [review]
1 nit.
Could you also add a licesning statement for CC0 like you did in the sister PR?

**Comment by mzxrules** [comment]
Changes made in this pr are licensed under CC0.

**Review by hensldm — APPROVED** [review]

**Review by Dragorn421 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**